### PR TITLE
Split admin and public apps

### PR DESCRIPTION
**Pull request by rail**
* Remove staging and testing environments.
* Use the `dev` branch as default for development.
* Build separate docker images for admin and public apps. Use separate
  requirements files, Dockerfile, but the same setup.py.
* Update README.rst to reflect the current deployment scenario.
* Refresh dependencies.
* Move public code to a separate package in the same namespace
  (shipit_api).
* Extract some model methods into functions, so the models can be used
  by the public app without methods that modify the database.
* Move `Product` and `ProductCategory` enums to a separate module.

### CODE REVIEW

**Comment by rail** [comment]
It was a bit hard to split this into logical commits without changing everything to pass CI.

**Comment by rail** [comment]
No rush with the review, I won't be landing this until after the release is out next week.

File: .taskcluster.yml [hunk 1]
```diff
@@ -22,17 +22,19 @@ tasks:
         # but the other environments do.
         $if: 'release_channel == "production"'
         then: 'relengstatic-prod-shipitfrontend-static-website'
-        else: 'relengstatic-${release_channel}-shipitfrontend-static-website'
+        else:
+          $if: 'release_channel == "dev"'
+          # TODO: switch to dev
```
Comment (mozbhearsum) [inline .taskcluster.yml]: Can we address this soon, please? :)

Comment (rail) [reply .taskcluster.yml]: Yes, I didn't want to do this at the same time. Once I get this deployed, I'm going to polish everything up!

**Pull request merged**